src/hotspot/share/cds/archiveBuilder.hpp

```diff
@@ -200,6 +200,7 @@ class ArchiveBuilder : public StackObj {
   SourceObjList _rw_src_objs;                 // objs to put in rw region
   SourceObjList _ro_src_objs;                 // objs to put in ro region
   ResizeableResourceHashtable<address, SourceObjInfo, ResourceObj::C_HEAP, mtClassShared> _src_obj_table;
+  ResizeableResourceHashtable<address, address, ResourceObj::C_HEAP, mtClassShared> _dumped_to_src_obj_table;
```

Comment:
I was trying to sort out these tables.  Can you have a comment above these tables what they are mapping? The names are somewhat opaque.

Reply:
I think people will first reach these functions when reading the code, so I added some comments there:

```
  // + When creating a CDS archive, we first load Java classes and create metadata
  //   objects as usual. These are call "source" objects.
  // + We then copy the source objects into the output buffer at "dumped addrsses".
  //
  // The following functions translate between these two (non-overlapping) spaces.
  // (The API should be renamed to be less confusing!)
  address get_dumped_addr(address src_obj) const;
  address get_src_obj(address dumped_addr) const;
```

I have filed an RFE [JDK-8292225](https://bugs.openjdk.org/browse/JDK-8292225) to clean up the names to make the core easier to understand.

Reply:
Ok, yes thanks.  but addrsses is missing an 'e' above.  Thanks for filing an additional bug so that one can guess what the tables mean even without knowing all the context.